### ISSUE TITLE
Upgrade to Vert.x 5

### ISSUE THREAD

**Issue by tsegismont** [reactiverse/reactiverse-contextual-logging#132]
- Vert.x 5 dependency upgrade
- JDK 11 minimum required version
- `ContextLocal` storage for better performances

**Comment by sishbi** [comment]
Hello, Please can you tell me when the new version will be available on Maven Central ?

**Comment by sishbi** [comment]
We use log4j2 so I guess this is correct?
https://mvnrepository.com/artifact/io.reactiverse/reactiverse-contextual-logging-log4j2/2.0.0

We previously just used:
```io.reactiverse:reactiverse-contextual-logging:1.2.1```

So there is a migration required for people who want to adopt `2.0.0`

**Comment by tsegismont** [comment]
Right. See https://reactiverse.io/reactiverse-contextual-logging/#_migrating_from_version_1
